### PR TITLE
Refactor: Remove adding transaction Id in request header

### PR DESCRIPTION
**Pull request by keyur9**
Refactor: Remove adding transaction Id in request header
Refactor: Add X-Kong-PID in response header
Refactor: Add sent success and sent failure count
Refactor: Timer for scheduling background thread
Add: max_events_sent_per_callback config option
Bump version to 0.2.22

### CODE REVIEW

File: kong/plugins/moesif/log.lua [hunk 1]
```diff
@@ -241,10 +246,28 @@ local function send_events_batch(premature)
         end
         if type(send_events_socket) == "table" and next(send_events_socket) ~= nil then
           local counter = 0
+          local total_events_sent_in_this_cycle = 0
```
Comment (dkm199) [inline kong/plugins/moesif/log.lua]: move this to line 232.

File: kong/plugins/moesif/log.lua [hunk 2]
```diff
@@ -241,10 +246,28 @@ local function send_events_batch(premature)
...
+               if pcall(send_payload, send_events_socket, parsed_url, batch_events, configuration) then 
+                sent_event = sent_event + #batch_events
+               end
+               local end_pay_time = socket.gettime()*1000
+               if configuration.debug then
+                ngx_log(ngx.DEBUG, "[moesif] send payload with event count - " .. tostring(#batch_events) .. " took time - ".. tostring(end_pay_time - start_pay_time).." for pid - ".. ngx.worker.pid())
+               end
+               batch_events = {}
+               ngx_log(ngx.DEBUG, "[moesif] Max events sent per callback, skip sending more events - " .." for pid - ".. ngx.worker.pid())
+               break
```
Comment (dkm199) [inline kong/plugins/moesif/log.lua:267]: this break only breaks this loop.. So at line 238, check that total_events_sent_in_this_cycle< configuration.max_events_sent_per_callback

**Review by dkm199 — COMMENTED** [review]
couple of comments, rest looks good to me

File: kong/plugins/moesif/moesif_ser.lua [hunk 3]
```diff
@@ -141,9 +142,12 @@ function _M.serialize(ngx, conf)
 
   -- Add Transaction Id to the response header
   if not conf.disable_transaction_id and transaction_id ~= nil then
-    response_headers["X-Moesif-Transaction-Id"] = generated_uuid
+    response_headers["X-Moesif-Transaction-Id"] = helper.uuid()
```
Comment (dgilling) [inline kong/plugins/moesif/moesif_ser.lua]: Don't override if already present in headers.

File: kong/plugins/moesif/schema.lua [hunk 4]
```diff
@@ -24,5 +24,6 @@ return {
     company_id_header = {default = "", type = "string"},
     max_callback_time_spent = {default = 1000, type = "number"},
     max_body_size_limit = {default = 500000, type = "number"},
+    max_events_sent_per_callback = {default = 100, type = "number"},
```
Comment (dgilling) [inline kong/plugins/moesif/schema.lua]: change to 400

**Comment by dkm199** [comment]
looks good

**Pull request merged**